vault/src/error.rs

```diff
@@ -44,10 +44,29 @@ pub enum Error {
     FromUtf8Error(#[from] FromUtf8Error),
     #[error("BroadcastStreamRecvError: {0}")]
     BroadcastStreamRecvError(#[from] BroadcastStreamRecvError),
+    #[error("Client has shutdown")]
+    ClientShutdown,
+    #[error("OsString parsing error")]
+    OsStringError,
+    #[error("File already exists")]
+    FileAlreadyExists,
+    #[error("There is a services already running on the system, with pid {0}")]
+    ServiceAlreadyRunning(u32),
+    #[error("Process with pid {0} not found")]
+    ProcessNotFound(String),
+    #[error("ParseIntError: {0}")]
+    ParseIntError(#[from] ParseIntError),
+    #[error("TokioError: {0}")]
+    TokioError(#[from] TokioJoinError),
+    #[error("System I/O error: {0}")]
+    IoError(#[from] IoError),
 }
 
-impl From<Error> for service::Error<Error> {
-    fn from(err: Error) -> Self {
-        Self::Retry(err)
```

Comment:
This is where we defaulted to `Retry`, is it possible to keep this for `Transient`?

Reply:
The above comment covers the solution, as the `From` trait for `Transient` error has already been implemented by the `backoff::Error` type.

Reply:
We need to default to `Transient` somehow, can you investigate how to do that?